consensus/types/src/light_client_finality_update.rs

```diff
@@ -1,69 +1,159 @@
+        finalized_block: &SignedBeaconBlock<E>,
+        finality_branch: FixedVector<Hash256, FinalizedRootProofLen>,
+        sync_aggregate: SyncAggregate<E>,
+        signature_slot: Slot,
+        fork_name: ForkName,
+    ) -> Result<Self, Error> {
+        let finality_update = match fork_name {
+            ForkName::Altair | ForkName::Merge => {
+                let finality_update = LightClientFinalityUpdateAltair {
+                    attested_header: LightClientHeaderAltair::block_to_light_client_header(
+                        attested_block,
+                    )?,
+                    finalized_header: LightClientHeaderAltair::block_to_light_client_header(
+                        finalized_block,
+                    )?,
+                    finality_branch,
+                    sync_aggregate,
+                    signature_slot,
+                };
+                Self::Altair(finality_update)
+            }
+            ForkName::Capella => {
+                let finality_update = LightClientFinalityUpdateCapella {
+                    attested_header: LightClientHeaderCapella::block_to_light_client_header(
+                        attested_block,
+                    )?,
+                    finalized_header: LightClientHeaderCapella::block_to_light_client_header(
+                        finalized_block,
+                    )?,
+                    finality_branch,
+                    sync_aggregate,
+                    signature_slot,
+                };
+                Self::Capella(finality_update)
+            }
+            ForkName::Deneb => {
+                let finality_update = LightClientFinalityUpdateDeneb {
+                    attested_header: LightClientHeaderDeneb::block_to_light_client_header(
+                        attested_block,
+                    )?,
+                    finalized_header: LightClientHeaderDeneb::block_to_light_client_header(
+                        finalized_block,
+                    )?,
+                    finality_branch,
+                    sync_aggregate,
+                    signature_slot,
+                };
+                Self::Deneb(finality_update)
+            }
+            ForkName::Base => return Err(Error::AltairForkNotActive),
+        };
 
-        let attested_header = decoder
-            .decode_next_with(|bytes| LightClientHeader::from_ssz_bytes(bytes, fork_name))?;
-        let finalized_header = decoder
-            .decode_next_with(|bytes| LightClientHeader::from_ssz_bytes(bytes, fork_name))?;
-        let finality_branch = decoder.decode_next()?;
-        let sync_aggregate = decoder.decode_next()?;
-        let signature_slot = decoder.decode_next()?;
+        Ok(finality_update)
+    }
 
-        Ok(Self {
-            attested_header,
-            finalized_header,
-            finality_branch,
-            sync_aggregate,
-            signature_slot,
+    pub fn get_slot<'a>(&'a self) -> Slot {
```

Comment:
This method name is ambiguous as we have `signature_slot` and `finalized_header.beacon.slot` as well